### PR TITLE
[Cookbook] - fix component_architecture

### PR DESCRIPTION
**Pull request by WebMamba**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| Issues        | Fix 
| License       | MIT

### CODE REVIEW

**Comment by smnandre** [comment]
Some things i noticed while coding the render:

The last code block will not work (wrong data attribute).

Some name with incorrect case (Symfony, UX, JS)

Some these/those i think.

The example with Header/Body to change, as they are slots/blocks and no isolated component (maybe use a button with an Icon inside?)

I'll make another review before monday :)

File: ux.symfony.com/cookbook/component_architecture.md [hunk 1]
```diff
@@ -3,17 +3,17 @@ title: Component architecture
...
 ---
 
 ## Introduction
 
-In SymfonyUX exist two packages: [TwigComponents](https://symfony.com/bundles/ux-twig-component/current/index.html) and [LiveComponent](https://symfony.com/bundles/ux-live-component/current/index.html).
+In Symfony UX exist two packages: [TwigComponents](https://symfony.com/bundles/ux-twig-component/current/index.html) and [LiveComponent](https://symfony.com/bundles/ux-live-component/current/index.html).
 Those two packages allow you to create reusable components in your Symfony application.
-But the component architecture is not exclusive to Symfony, it is a design pattern that can be applied to any programming language or framework.
+But the component architecture is not exclusive to Symfony, it's a design pattern that can be applied to any programming language or framework.
 And the js world already implement this architecture for long time, on many different frameworks like React, Vue, or Svelte.
```
Comment (Kocal) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
And the JavaScript world already implements this architecture for long time, on many different frameworks like React, Vue, or Svelte.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 2]
```diff
@@ -50,18 +50,18 @@ Or you can make composition with the following syntax:
 </twig:Card>
 ```
 
-So here we Card component, and we give to the content of this component mutliple other components.
+So here we Card component, and we give to the content of this component two others components.
```
Comment (Kocal) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
So here we have a Card component, and we give to the content of this component two others components.
```

Not sure about the 1st sentence tho

File: ux.symfony.com/cookbook/component_architecture.md [hunk 3]
```diff
@@ -50,18 +50,18 @@ Or you can make composition with the following syntax:
...
 
 ### Independence
 
 This is a really important rule, and not obvious. But your component should leave on his own context,
-he should not be aware of the rest of the page. You should to talk one component into a page, to another and it should work exactly the same.
-This rule make your component trully reusable.
+he should not be aware of the rest of the page. You should be able to take one component into a page, to another and it should work exactly the same.
```
Comment (Kocal) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
it should not be aware of the rest of the page. You should be able to take one component into a page, to another and it should works exactly the same.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 4]
```diff
@@ -50,18 +50,18 @@ Or you can make composition with the following syntax:
...
+This rule make your component truly reusable.
 
 ***How does it work into Symfony?***
 
-Symfony keep the context of the page into the context of your component. So this your own responsability to follow this rules.
-But notice that if there are conflic between a variable from the context page and your component, your component context override the page context.
+Symfony keep the context of the page into the context of your component. So this your own responsibility to follow this rules.
```
Comment (Kocal) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
Symfony keeps the context of the page into the context of your component. So this your own responsibility to follow those rules.
```

(or "this rule")

File: ux.symfony.com/cookbook/component_architecture.md [hunk 5]
```diff
@@ -50,18 +50,18 @@ Or you can make composition with the following syntax:
...
+But notice that if there are conflict between a variable from the context page and your component, your component context override the page context.
```
Comment (Kocal) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
But notice that if there are conflicts between a variable from the context page and your component, your component context override the page context.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 6]
```diff
@@ -117,7 +117,7 @@ And when the loading is done, the state `loading` can be set to `false` and the
 
 ***How does it work into Symfony?***
 
-In symfony you 2 different approach to handle state. The first one is to use stimulus directly
+In Symfony you 2 different approach to handle state. The first one is to use stimulus directly
```
Comment (Kocal) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
In Symfony you have two different approaches to handle state. The first one is to use stimulus directly
```

**Comment by Kocal** [comment]
I believe my comments can be replaced by an automatic tool, do you think we can use https://github.com/get-alex/alex or https://github.com/textlint/textlint or https://github.com/huacnlee/autocorrect or anything better?

**Review by Xbirdfr — APPROVED** [review]

**Comment by WebMamba** [comment]
Thanks a lot, @Kocal! I am so bad at spelling 😭

File: ux.symfony.com/cookbook/component_architecture.md [hunk 7]
```diff
@@ -41,27 +41,25 @@ Or you can make composition with the following syntax:
...
 
-So here we Card component, and we give to the content of this component mutliple other components.
+So here we have a Card component, and we give to the content of this component two others components.
 
 ### Independence
 
 This is a really important rule, and not obvious. But your component should leave on his own context,
-he should not be aware of the rest of the page. You should to talk one component into a page, to another and it should work exactly the same.
-This rule make your component trully reusable.
+it should not be aware of the rest of the page. You should be able to take one component into a page, to another and it should works exactly the same.
```
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
it should not be aware of the rest of the page. You should be able to take one component into a page, to another and it should work exactly the same.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 8]
```diff
@@ -41,27 +41,25 @@ Or you can make composition with the following syntax:
...
+    </twig:Button>
 </twig:Card>
 ```
 
-So here we Card component, and we give to the content of this component mutliple other components.
+So here we have a Card component, and we give to the content of this component two others components.
 
 ### Independence
 
 This is a really important rule, and not obvious. But your component should leave on his own context,
```
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
This is a really important rule, and not obvious. But your component should live on his own context,
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 9]
```diff
@@ -41,27 +41,25 @@ Or you can make composition with the following syntax:
...
-    </twig:CardBody>
+    <twig:Icon name="info"/>
+    <twig:Button>
+        <twig:Icon name="close" />
+    </twig:Button>
 </twig:Card>
 ```
 
-So here we Card component, and we give to the content of this component mutliple other components.
+So here we have a Card component, and we give to the content of this component two others components.
```
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
So here we have a Card component, and we give to the content of this component two other components.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 10]
```diff
@@ -41,27 +41,25 @@ Or you can make composition with the following syntax:
...
-This rule make your component trully reusable.
+it should not be aware of the rest of the page. You should be able to take one component into a page, to another and it should works exactly the same.
+This rule make your component truly reusable.
 
 ***How does it work into Symfony?***
 
-Symfony keep the context of the page into the context of your component. So this your own responsability to follow this rules.
-But notice that if there are conflic between a variable from the context page and your component, your component context override the page context.
+Symfony keeps the context of the page into the context of your component. So this your own responsibility to follow those rules.
+But notice that if there are conflicts between a variable from the context page and your component, your component context override the page context.
```
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
But notice that if there are conflicts between a variable from the context page and your component, your component context overrides the page context.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 11]
```diff
@@ -3,17 +3,17 @@ title: Component architecture
...
+In Symfony UX exist two packages: [TwigComponents](https://symfony.com/bundles/ux-twig-component/current/index.html) and [LiveComponent](https://symfony.com/bundles/ux-live-component/current/index.html).
 Those two packages allow you to create reusable components in your Symfony application.
-But the component architecture is not exclusive to Symfony, it is a design pattern that can be applied to any programming language or framework.
-And the js world already implement this architecture for long time, on many different frameworks like React, Vue, or Svelte.
-So, a set of rules and pattern has already be defined to work with components. This is why SymfonyUX try to be as close as possible to those rules.
+But the component architecture is not exclusive to Symfony, it's a design pattern that can be applied to any programming language or framework.
+And the Javascript world already implements this architecture for long time, on many different frameworks like React, Vue, or Svelte.
+So, a set of rules and pattern has already be defined to work with components. This is why Symfony UX try to be as close as possible to those rules.
```
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
So, a set of rules and patterns has already be defined to work with components. This is why Symfony UX tries to be as close as possible to those rules.
```

Comment (smnandre) [reply ux.symfony.com/cookbook/component_architecture.md]: (and just after:

```diff
- So let's see what are those rules!
+ So, let's see what those rules are!
```

File: ux.symfony.com/cookbook/component_architecture.md (hunk 7 above)
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
it should not be aware of the rest of the page. You should be able to take a component from a page to another and it should works exactly the same.
```

File: ux.symfony.com/cookbook/component_architecture.md [hunk 12]
```diff
@@ -41,27 +41,25 @@ Or you can make composition with the following syntax:
...
-So here we Card component, and we give to the content of this component mutliple other components.
+So here we have a Card component, and we give to the content of this component two others components.
 
 ### Independence
 
 This is a really important rule, and not obvious. But your component should leave on his own context,
-he should not be aware of the rest of the page. You should to talk one component into a page, to another and it should work exactly the same.
-This rule make your component trully reusable.
+it should not be aware of the rest of the page. You should be able to take one component into a page, to another and it should works exactly the same.
+This rule make your component truly reusable.
```
Comment (smnandre) [inline ux.symfony.com/cookbook/component_architecture.md]:
```suggestion
This rule makes your component truly reusable.
```

**Comment by WebMamba** [comment]
Thanks a lot @smnandre! All fixed!

**Comment by smnandre** [comment]
Thanks you !! Let's merge this and we can iterate next to add features, tags, etc etc !! 

👏

**Review by smnandre — APPROVED** [review]

**Comment by kbond** [comment]
Thanks Matheo.

**Pull request merged**